### ISSUE TITLE
Investigate using new `rnn_vad` from webrtc upstream

### ISSUE THREAD

**Issue by daanzu** [daanzu/py-webrtcvad-wheels#2]
* modules/audio_processing/agc2/rnn_vad - external/webrtc - Git at Google (https://chromium.googlesource.com/external/webrtc/+/3a99ebefbb5e3faf33055a63e11eadb69c6ddc6f/modules/audio_processing/agc2/rnn_vad/)

* jzi040941/py-webrtcrnnvad: Python interface to the RNNoise VAD(Voice Activity Detection) component inside webrtc (https://github.com/jzi040941/py-webrtcrnnvad)
* jzi040941/webrtc_rnnvad: webrtc_rnnvad (https://github.com/jzi040941/webrtc_rnnvad)
* WickrInc/rnnoise: Recurrent neural network for audio noise reduction (https://github.com/WickrInc/rnnoise)
* PulseAudio / webrtc-audio-processing · GitLab (https://gitlab.freedesktop.org/pulseaudio/webrtc-audio-processing/)
* Using rnn_vad in my project (https://groups.google.com/g/discuss-webrtc/c/b7PlW_M0Qks)

* src - Git at Google (https://webrtc.googlesource.com/src/)

**Comment by sandsc** [comment]
Agreed—this feature is greatly needed. RNN VAD can recognize noise and reduce the probability of false positives.

**Comment by daanzu** [comment]
I'm unsure if it's worth it, given the accuracy of silero-vad for a roughly similar computational cost: https://github.com/daanzu/voice-activity-benchmark

**Comment by m13v** [comment]
the rnn_vad from webrtc is significantly better than the classic GMM-based VAD, especially for noisy environments. we tested both for a voice-first desktop agent and the RNN version has way fewer false positives on keyboard sounds and fan noise. the latency is also lower since it works on smaller frames.

the main challenge with wrapping it in python is that the rnn_vad module has more dependencies within the webrtc tree than the old vad. jzi040941's webrtc_rnnvad repo is a good starting point but it's pinned to an older webrtc revision. might be worth extracting just the rnn_vad subfolder and its dependencies rather than pulling the whole tree.

**Comment by m13v** [comment]
we use VAD extensively in our voice-controlled macOS agent for detecting speech boundaries. the transcription pipeline with VAD integration: https://github.com/m13v/fazm/blob/main/Desktop/Sources/TranscriptionService.swift

and the floating UI that reacts to voice activity state: https://github.com/m13v/fazm/blob/main/Desktop/Sources/FloatingControlBar/FloatingControlBarView.swift
